File: problem/3/perl/solve_3.pl
#!/usr/bin/env perl6
# http://projecteuler.net/problem=3

use v6;

my $p = 2;
my $N = @*ARGS[0];
my @factors := gather while $N >= $p**2 {
        if $N % $p == 0 {
                take $p;
                $N /= $p;
        } else {
                $p++;
        }
}
@factors.push($N);

say [max] @factors;

File: problem/3/python/solve_3.py
#!/usr/bin/env python
# http://projecteuler.net/problem=3

import sys

N = int(sys.argv[1])
p = 2
factors = []

while N >= p**2:
        if N % p == 0:
                factors.append(p)
                N /= p
        else:
                p = p + 1
factors.append(N)

print max(factors)
print factors
